out-lua/Makefile

```diff
@@ -1,4 +1,7 @@
 CC ?= clang
+USER ?= root
+RUNHOST ?= localhost
+RUNDIR ?= $(HOME)
```

Comment:
This will be `$HOME` on the local, not remote, machine. Something like `RUNDIR?=ssh $(SSH_OPTIONS) -P $(SSHPORT) $(USER)@$(RUNHOST) pwd` is probably better?

Reply:
This is run with `... copy-lua` and `... exec-tests` and it has been tested successfully on a remote machine (morello). However, I can do what you suggest if it makes things better/easier to understand.

Reply:
But you are right. That should not work... I'll investigate. Thanks.

Reply:
> This will be `$HOME` on the local, not remote, machine. Something like `RUNDIR?=ssh $(SSH_OPTIONS) -P $(SSHPORT) $(USER)@$(RUNHOST) pwd` is probably better?

Interestingly, there is no way to do this except by setting `SHELL` as `ssh user@host` and then run commands. However, I do need both `RUNHOST` and `RUNDIR` BUT `RUNHOST` must be resolved locally. I want to keep `RUNHOST` as local name to avoid to expose the IP address in the external config (`.buildbot.sh`).

Reply:
Another solution would be to pass `RUNDIR` and set its default value as `/root`.

Reply:
> Interestingly, there is no way to do this except by setting SHELL as ssh user@host and then run commands. 

Not sure I understand.

Reply:
Looking a bit more at this: can't we just drop `RUNDIR` entirely and assume the user is running this in `pwd` and that's OK?

Reply:
> > Interestingly, there is no way to do this except by setting SHELL as ssh user@host and then run commands.
> 
> Not sure I understand.

https://stackoverflow.com/questions/65633894/gnu-make-remote-command-with-ssh-as-the-shell.

I can't get the output of `ssh ... pwd`, or, at least, I couldn't find how to...

Reply:
`pwd` is a normal shell command, so just `ssh <host> pwd` will work. [You can then use whatever SSH options you want with that command.]

Reply:
> Looking a bit more at this: can't we just drop `RUNDIR` entirely and assume the user is running this in `pwd` and that's OK?

If you look at the [Makefile](https://github.com/capablevms/llvm-function-split/pull/36/files#diff-a49aaf0990b0d30c7639f09bd0ab057c2970876d113fc9c75b9761b89d2c3999L11). Now, we hardcode it to `/root/lua` and it works inside `qemu`. However, to give flexibility we have to set `rpath` somehow.

Reply:
I think my confusion is that I understand why `copy-lua` uses RUNDIR but I don't understand why `cc` (which will be run as a cross-compiler in non-CHERI I think?) also uses this:

```
$(CC) $(CFLAGS) $< -L. -L$(RUNDIR)/lua $(patsubst %.bc,-l%,$(filter-out _main.bc,$(wildcard *.bc))) -L. -L$(RUNDIR)/lua -ldl -lm -g -Wl,-rpath,$(RUNDIR)/lua -o lua.shared
```

Reply:
> I think my confusion is that I understand why `copy-lua` uses RUNDIR but I don't understand why `cc` (which will be run as a cross-compiler in non-CHERI I think?) also uses this:
> 
> ```
> $(CC) $(CFLAGS) $< -L. -L$(RUNDIR)/lua $(patsubst %.bc,-l%,$(filter-out _main.bc,$(wildcard *.bc))) -L. -L$(RUNDIR)/lua -ldl -lm -g -Wl,-rpath,$(RUNDIR)/lua -o lua.shared
> ```

Without `rpath`, there will be errors about missing shared objects. That has always been hardcoded, unfortunately. That line without `CFLAGS` can be run without cross-compiling.